### ISSUE TITLE
Uncaught failure when attempting to install hpsspy

### ISSUE THREAD

**Issue by weaverba137** [desihub/desiconda#66]
This issue is to document two issues, first with the way `hpsspy` is installed, and second with the sourced script that attempts to perform the installation.

1. In `conf/pip-pkgs.sh`, [Line 4](https://github.com/desihub/desiconda/blob/e34365d7313e29f8923af0a5cf57dca22462a0a6/conf/pip-pkgs.sh#L4), `hpsspy` is installed with an unusual `--no-binary :all:` option, which is odd, since `hpsspy` is pure Python. The file appears to have been created by @marcelo-alvarez. Could you please explain why `hpsspy` is installed this way?
   - In fact `--no-binary :all:` doesn't work, which leaves `hpsspy` uninstalled. See below for error messages
   - Installing `hpsspy` with an ordinary install appears to work fine.  See below.
2. `conf/pip-pgkgs.sh` only checks the status of the final pip command, so if and when the installation of `hpsspy` fails, `conf/pip-pkgs.sh will` just keep running.  It only detects failure to install `mpi4py`.  So the installation of `threadpoolctl` could *also* fail, and we wouldn't know it.  `set -e` should be set for the entire top-level `install.sh` script, and tested to make sure that failures do in fact cause the install to crash.

Installation logs:

`desi@login28[1027]$ pip install --dry-run --no-binary :all: hpsspy`:
```
DEPRECATION: --no-binary currently disables reading from the cache of locally built wheels. In the future --no-binary will not influence the wheel cache. pip 23.1 will enforce this behaviour change. A possible replacement is to use the --no-cache-dir option. You can use the flag --use-feature=no-binary-enable-wheel-cache to test the upcoming behaviour. Discussion can be found at https://github.com/pypa/pip/issues/11453
Collecting hpsspy
  Using cached hpsspy-0.6.1.tar.gz (39 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Traceback (most recent call last):
        File "/global/common/software/desi/perlmutter/desiconda/20230111-2.1.0/conda/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 351, in <module>
          main()
        File "/global/common/software/desi/perlmutter/desiconda/20230111-2.1.0/conda/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 333, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/global/common/software/desi/perlmutter/desiconda/20230111-2.1.0/conda/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-jbmxplzz/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/tmp/pip-build-env-jbmxplzz/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-jbmxplzz/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 338, in run_setup
          exec(code, locals())
        File "<string>", line 50, in <module>
        File "/global/common/software/desi/perlmutter/desiconda/20230111-2.1.0/conda/lib/python3.10/importlib/__init__.py", line 126, in import_module
          return _bootstrap._gcd_import(name[level:], package, level)
        File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
      ModuleNotFoundError: No module named 'hpsspy'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

`desi@login28[1028]$ pip install --dry-run hpsspy`
```
Collecting hpsspy
  Using cached hpsspy-0.6.1-py3-none-any.whl (31 kB)
Requirement already satisfied: pytz in /global/common/software/desi/perlmutter/desiconda/20230111-2.1.0/conda/lib/python3.10/site-packages (from hpsspy) (2022.7)
Would install hpsspy-0.6.1
```

**Comment by weaverba137** [comment]
Pinging @marcelo-alvarez.

**Comment by weaverba137** [comment]
Closed by #68.
